integreat_cms/static/src/js/machine-translation-overlay.ts

```diff
@@ -0,0 +1,216 @@
+    overlay.classList.add("hidden");
+    overlay.classList.remove("flex");
+};
+
+const toggleOptionalText = (trigger: HTMLElement, optionalText: HTMLElement) => {
+    // function to toggle optional text
+    trigger.addEventListener("click", () => {
+        optionalText.classList.toggle("hidden");
+        optionalText.classList.toggle("block");
+        if (optionalText.classList.contains("block")) {
+            trigger.textContent = trigger.dataset.alternativeText; // eslint-disable-line no-param-reassign
+        } else {
+            trigger.textContent = trigger.dataset.defaultText; // eslint-disable-line no-param-reassign
+        }
+    });
+};
+
+const addMachineTranslationOverlayEventListeners = () => {
+    // add event listeners to overlay
+    const overlay = document.getElementById("machine-translation-overlay");
+    const provider = document.getElementById("machine-translation-option").getAttribute("data-mt-provider");
+    if (overlay && provider === "DeepL") {
+        // Set listener for bulk action execute button.
+        document.getElementById("bulk-action-execute")?.addEventListener("click", (event) => {
+            const machineTranslationOptionIndex = (
+                document.getElementById("machine-translation-option") as HTMLOptionElement
+            ).index;
+            const { selectedIndex } = document.getElementById("bulk-action") as HTMLSelectElement;
+
+            if (machineTranslationOptionIndex === selectedIndex) {
+                const url = (document.getElementById("machine-translation-option") as HTMLElement).dataset.url;
+                const selectedIDs = getSelectedIDs();
+                const filteredPages = getAllPages(url, selectedIDs);
+
+                event.preventDefault();
+                overlay.classList.remove("hidden");
+                overlay.classList.add("flex");
+
+                filteredPages.then((filteredPages: Pages) => {
+                    prepareOverlay(filteredPages);
+                });
+
+                toggleOptionalText(
+                    document.getElementById("machine-translation-overlay-expansion-trigger"),
+                    document.getElementById("machine-translation-overlay-pages-optional")
+                );
+                toggleOptionalText(
+                    document.getElementById("machine-translation-overlay-warning-more"),
+                    document.getElementById("machine-translation-overlay-warning-optional")
+                );
+                setAmountOfSelectedPages(selectedIDs.length);
+            }
+        });
+        document.getElementById("btn-close-machine-translation-overlay").addEventListener("click", () => {
+            closeMachineTranslationOverlay(overlay);
+        });
+        // Close window by clicking on backdrop.
+        overlay.addEventListener("click", (e) => {
+            if (e.target === overlay) {
+                closeMachineTranslationOverlay(overlay);
+            }
+        });
+    }
```

Comment:
```suggestion
                
                setAmountOfSelectedPages(selectedIDs.length);
            }
        });
        document.getElementById("btn-close-machine-translation-overlay").addEventListener("click", () => {
            closeMachineTranslationOverlay(overlay);
        });
        // Close window by clicking on backdrop.
        overlay.addEventListener("click", (e) => {
            if (e.target === overlay) {
                closeMachineTranslationOverlay(overlay);
            }
        });

        toggleOptionalText(
            document.getElementById("machine-translation-overlay-warning-more"),
            document.getElementById("machine-translation-overlay-warning-optional")
        );
        toggleOptionalText(
            document.getElementById("machine-translation-overlay-expansion-trigger"),
            document.getElementById("machine-translation-overlay-pages-optional")
        );
    }
```

Event listner on "Show pages" were piled up and click toggle didn't work sometimes.